### PR TITLE
Session6 uploaded

### PR DESCRIPTION
**Pull request by meta-aman-gautam**

### CODE REVIEW

File: Session6/Polynomial_Operations/Poly.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
+package Polynomial_Operations;
+
+import java.util.Arrays;
+
+public class Poly {
+
+	public int[] polynomial;
```
Comment (meta-prateek-jain) [inline Session6/Polynomial_Operations/Poly.java]: Why public ?

File: Session6/Polynomial_Operations/Poly.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
+package Polynomial_Operations;
+
+import java.util.Arrays;
+
```
Comment (meta-prateek-jain) [inline Session6/Polynomial_Operations/Poly.java]: Class level comments?

File: Session6/Polynomial_Operations/Poly.java [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+	 * @param inputPolynomial is input polynomial from user
+	 */
+	public Poly(int[] inputPolynomial) {
+		polynomial = inputPolynomial;
+		if (polynomial.length == 0)
+			throw new AssertionError("Array empty");
+	}
+
+	public Poly() {
+		// default constructor for Poly
```
Comment (meta-prateek-jain) [inline Session6/Polynomial_Operations/Poly.java]: Why empty constructor?

File: Session6/Polynomial_Operations/Poly.java [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+	 *            is first Polynomial
+	 * @param value2
+	 *            is second polynomial
+	 * @return String value of polynomial
+	 */
+	public String mul(Poly value1, Poly value2) {
+		int length1 = value1.polynomial.length;
+		int length2 = value2.polynomial.length;
+		int mulList[] = new int[length1 + length2 - 1];
+		String retString = null;
```
Comment (meta-prateek-jain) [inline Session6/Polynomial_Operations/Poly.java]: Why initializing with null value?

File: Session6/Polynomial_Operations/PolyTest.java [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+
+@RunWith(Parameterized.class)
+public class PolyTest {
+	
+	public Poly Poly =new Poly();
```
Comment (meta-prateek-jain) [inline Session6/Polynomial_Operations/PolyTest.java]: Why public?

File: Session6/Polynomial_Operations/PolyTest.java [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+		// is test of multiply of Polynomial
+		this.evaluatedValueMultiply = evaluatedValueMultiply;
+		this.mul1 =mul1;
+		this.mul2 =mul2;
+	}
+
+	@Parameterized.Parameters
+	public static Collection data() {
+		return Arrays.asList(new Object[][] { 
+			{  new Poly(new int[] { 6, 5, 2 }).degree(), 2,
```
Comment (meta-prateek-jain) [inline Session6/Polynomial_Operations/PolyTest.java]: Here we are only supposed to pass values directly. We don't call the function over here

File: Session6/Polynomial_Operations/PolyTest.java [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+import java.util.Arrays;
+import java.util.Collection;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+
+@RunWith(Parameterized.class)
+public class PolyTest {
```
Comment (meta-prateek-jain) [inline Session6/Polynomial_Operations/PolyTest.java]: Update this test class with proper test methods

File: Session6/set_Operations/IntSet.java [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+			if (union.contains(set1.set[count])) {
+				continue;
+			}
+			union.add(set1.set[count]);
+		}
+
+		for (int count = 0; count < set2.set.length; count++) {
+			if (union.contains(set2.set[count])) {
+				continue;
+			} else{
```
Comment (meta-prateek-jain) [inline Session6/set_Operations/IntSet.java]: Optimize the code. This functionality can be achieved without using **continue**

File: Session6/set_Operations/IntSetTest.java [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+import java.lang.Character.Subset;
+import java.util.Arrays;
+import java.util.Collection;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+@RunWith(Parameterized.class)
+public class IntSetTest {
```
Comment (meta-prateek-jain) [inline Session6/set_Operations/IntSetTest.java]: Update this test class as well

File: Session6/Polynomial_Operations/PolyTest.java (hunk 7 above)
Comment (meta-aman-gautam) [reply Session6/Polynomial_Operations/PolyTest.java]: Sir , please explain this ?

Comment (meta-prateek-jain) [reply Session6/Polynomial_Operations/PolyTest.java]: Include Negative and positive test case both.

**Pull request merged**